src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -131,40 +129,34 @@ public void setInOvertime(boolean o) {
 			inOvertime = o;
 			scoreBoardChange(new ScoreBoardEvent(this, EVENT_IN_OVERTIME, new Boolean(inOvertime), last));
 		}
-		if (!o) {
-			try {
-				OvertimeLineupTimePolicy p = (OvertimeLineupTimePolicy)getPolicy(OvertimeLineupTimePolicy.ID);
-				if (null != p)
-					p.stopOvertime();
-			} catch ( ClassCastException ccE ) {
-				ScoreBoardManager.printMessage("Internal Error: invalid OvertimeLineupTimePolicy : "+ccE.getMessage());
-			}
+		ClockModel lc = getClockModel(Clock.ID_LINEUP);
+		if (!o && lc.isCountDirectionDown()) {
+			lc.setMaximumTime(settings.getLong("Clock." + Clock.ID_LINEUP + ".Time"));
 		}
 	}
 	public void startOvertime() {
 		synchronized (runLock) {
 			requestBatchStart();
 			ClockModel pc = getClockModel(Clock.ID_PERIOD);
 			ClockModel jc = getClockModel(Clock.ID_JAM);
+			ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
+			ClockModel lc = getClockModel(Clock.ID_LINEUP);
 			if (pc.isRunning() || jc.isRunning())
 				return;
 			if (pc.getNumber() < pc.getMaximumNumber())
 				return;
-			if (pc.getTime() > pc.getMinimumTime())
+			if (!pc.isTimeAtEnd())
 				return;
-			pc.setTime(1000);
+			pc.setTime(1000); //TODO: Should not be needed
 			setInPeriod(true);
 			setInOvertime(true);
-			getClockModel(Clock.ID_INTERMISSION).stop();
-			try {
-				OvertimeLineupTimePolicy p = (OvertimeLineupTimePolicy)getPolicy(OvertimeLineupTimePolicy.ID);
-				if (null != p)
-					p.startOvertime();
-			} catch ( ClassCastException ccE ) {
-				ScoreBoardManager.printMessage("Internal Error: invalid OvertimeLineupTimePolicy : "+ccE.getMessage());
+			ic.stop();
+			long otLineupTime = settings.getLong("Clock." + Clock.ID_LINEUP + ".OvertimeTime");
+			if (lc.getMaximumTime() < otLineupTime) {
```

Comment:
I don't understand this comment.

Reply:
Why is there an if here rather than being unconditional?

Reply:
Because we don't want the maximum time to be reduced. For starters it would break AutoStart, which needs a maximum time of at least lineup time + buffer time.